docker-compose/sonar-with-postgres/README.md

```diff
@@ -0,0 +1,219 @@
+***
+
+## **Pre-requisites**
+
+**To set up SonarQube with a Postgres database using docker-compose, you will need to have the following prerequisites installed and configured on your machine:**
+
+**Docker:** Docker is a containerization platform that allows you to run applications in lightweight, isolated environments called containers. You will need to have Docker installed on your machine to use docker-compose.
+
+**Docker Compose:** Docker Compose is a tool for defining and running multi-container Docker applications. You will use Docker Compose to set up and run the SonarQube and Postgres containers.
+
+**A text editor:** You will need a text editor to create and edit the docker-compose.yml file that defines the configuration for your SonarQube and Postgres containers. Some popular text editors include Visual Studio Code, Sublime Text, and Atom.
+
+**A Postgres database:** SonarQube requires a Postgres database to store its data. The Postgres database can be run in a separate container using Docker, or it can be installed and run locally on your machine.
+
+**A SonarQube license:** If you are using the SonarQube Community Edition, you do not need a license. However, if you are using the SonarQube Enterprise Edition, you will need to obtain a license from SonarSource.
+
+Once you have these prerequisites installed and configured, you can proceed with setting up SonarQube with a Postgres database using docker-compose.
+
+A prerequisite with respect to Elasticsearch is to keep your max virtual memory areas vm.max_map_count increase to at least [262144]
+To do that , if you are a windows user, Please run the following command in terminal. 
+
+**```wsl.exe -d docker-desktop sh -c "sysctl -w vm.max_map_count=262144"```**
+
+if you are a Linux user, Please run the following command in terminal. 
+
+**```sudo sysctl -w vm.max_map_count=262144```**
+
+***
+# **How to Setup SonarQube Server with Docker-compose?**
+Now, We will create a simple **docker-compose.yml** file and store the code in it.
+
+### **1. Open your Visual Studio Code.**
+![Visual studio page](images/vscodepage.png)
+***
+
+### **2. Create a folder , and name it by your choice.**
+**To create a Folder run the following command**
+
+```mkdir <folder-name>``` 
+
+![Create a Folder](images/openfolder.png)
+***
+
+### **3. Create docker-compose.yml .**
+
+```
+version: '3.7'
+services:
+  sonarqube:
+    image: sonarqube:latest
+    ports:
+      - 9000:9000
+    environment:
+      - SONARQUBE_JDBC_USERNAME=sonar
+      
+      - SONARQUBE_JDBC_PASSWORD=sonar
+      - SONARQUBE_JDBC_URL=jdbc:postgresql://db:5432/sonar
+    volumes:
+      - sonarqube_conf:/opt/sonarqube/conf
+      - sonarqube_data:/opt/sonarqube/data
+      - sonarqube_extensions:/opt/sonarqube/extensions
+      - sonarqube_bundled-plugins:/opt/sonarqube/lib/bundled-plugins
+    command: ["./bin/run.sh"]
```

Comment:
remove this line and run again to test, if everything is working. if yes - we dont need this line

Reply:
Yes removed it . and runned once again , is running succefully